bundler/lib/bundler/compact_index_client/cache.rb

```diff
@@ -55,14 +55,13 @@ def versions_etag_path
       end
 
       def checksums
-        checksums = {}
-
-        lines(versions_path).each do |line|
-          name, _, checksum = line.split(" ", 3)
-          checksums[name] = checksum
+        lines(versions_path).each_with_object({}) do |line, checksums|
+          name_end = line.index(" ")
+          checksum_start = line.index(" ", name_end + 1) + 1
+          
+          name = line[0, name_end]
+          checksums[name] = line[checksum_start..-1]
```

Comment:
```suggestion
          checksums[name] = line[checksum_start, line.size - checksum_start]
```

will avoid allocating a Range object